llvm/lib/IR/Verifier.cpp

```diff
@@ -5661,6 +5661,12 @@ void Verifier::visitIntrinsicCall(Intrinsic::ID ID, CallBase &Call) {
     }
     break;
   }
+  case Intrinsic::experimental_patchpoint: {
+    // Check that result type can fit in a single register.
+    Check(Call.getType()->isSingleValueType(),
```

Comment:
This isn't really what this means. It may be reasonable to assert this, but we have no way of knowing anything about the machine registers here

Reply:
I was basing this off the doc comment for `isSingleValueType`: "Return true if the type is a valid type for a register in codegen."
Do you have a suggestion for constraining the type somehow?

Reply:
I think the comment (on isSingleValueType)  is misleading and not helpful, but in this case means "doesn't require decomposing into multiple EVTs to represent". I would just drop the comment